test/parallel/test-async-wrap-GH13045.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+const common = require('../common');
+
+// Checks for the issue in Refs.
+// An HTTP Agent reuses a TLSSocket, and makes a failed call to `asyncReset`.
+
+const assert = require('assert');
+const https = require('https');
+const fs = require('fs');
+
+// arrow factory so fail point will appear in the stack
+function mustNotErr() {
```

Comment:
I don't understand why you need this. You can just pass `assert.fail` (without the parens) as the handler, everywhere you use `mustNotErr()`.